lambda_src/lambda_function.py

```diff
@@ -186,22 +193,26 @@ def lambda_handler(event: Any, context: Any) -> Dict[Text, Any]:
     """
     method = event.get('httpMethod')
     headers = event['headers']
-    LOG.debug(f'lambda_handler() called.')
+    CONSOLE_LOGGER.debug(f'lambda_handler() called.')
 
     destination = headers.get(DESTINATION_URI_HEADER)
     batch_id = headers.get(BATCH_ID_HEADER)
 
-    # httpMethod doesn't exist implies caller is base lambda.
-    # This is required to break an infinite loop of child lambda creation.
-    if not method:
-        return sync_flow(event, context)
-
-    # httpMethod exists implies caller is API Gateway
-    if method == 'POST' and destination:
-        return async_flow_init(event, context)
-    elif method == 'POST':
-        return sync_flow(event, context)
-    elif method == 'GET':
-        return async_flow_poll(destination, batch_id)
-
-    return create_response(400, 'Unexpected Request.')
+    try:
+        # httpMethod doesn't exist implies caller is base lambda.
+        # This is required to break an infinite loop of child lambda creation.
+        if not method:
+            return sync_flow(event, context)
+
+        # httpMethod exists implies caller is API Gateway
+        if method == 'POST' and destination:
+            return async_flow_init(event, context)
+        elif method == 'POST':
+            return sync_flow(event, context)
+        elif method == 'GET':
+            return async_flow_poll(destination, batch_id)
+
+        return create_response(400, 'Unexpected Request.')
+    except Exception as e:
+        GEFF_SENTRY_LOGGER.exception(e)
```

Comment:
what if SENTRY isn't set up in a deployment. not every user will use it.

Reply:
The logger is setup regardless of Sentry setup. Specific loggers are channeled into specific projects in Sentry. The `sentry_sdk` is initialized only if `SENTRY_DSN` and `GEFF_DSN` env variables are set.

Reply:
[Here](https://github.com/Snowflake-Labs/geff/blob/9b1c9b77730a10a63b66aac2540566a1f855e7b0/lambda_src/utils.py#L129-L159) is the definition of the `setup_sentry`.